src/App.tsx

```diff
@@ -1,61 +1,126 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { getUserById } from './services/user';
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [userIdError, setUserIdError] = useState(false);
+
+  const [todoList, setTodoList] = useState<Todo[]>(todos);
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setTitleError(false);
+  };
+
+  const handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+  };
+
+  const newId = Math.max(...todos.map((el) => el.id));
+
+  const newTodo = {
+    title,
+    userId,
+    completed: false,
+    user: getUserById(userId),
+    id: newId + 1,
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim()) {
+      setTitleError(true);
+    }
+
+    if (!userId) {
+      setUserIdError(true);
+    }
+
+    if (!userId || !title.trim()) {
```

Comment:
You need to use && here, because || will be completed when one of conditions will be true
```suggestion
    if (!userId && !title.trim()) {
```

Reply:
not vise versa? if i do like you told me  - i can add only one field . but if i use my aproach i cant add one field